rpc/connection.go

```diff
@@ -609,6 +609,9 @@ func (c *Connection) doReconnect(ctx context.Context, disconnectStatus Disconnec
 			// A fatal error happened.
 			*reconnectErrPtr = err
 			// short-circuit Retry
+			if err != nil {
+				c.transport.Close()
```

Comment:
Ah ok thanks.  I didn't realize you could `Dial` again on a closed transport.  So is the real issue that `Dial` doesn't close `t.conn` if it's non-nil, before overwriting it?  Are the other ways to call `Dial` on the same transport twice?